pkg/noderesourcetopology/TESTS.md

```diff
@@ -0,0 +1,77 @@
+
+### Table guidelines:
+- All pods are ```Guaranteed``` QoS pods, IOW all containers are asking for an equal ```request``` and ```limit``` values.
+- The table describes the coverage of the ```TopologyManagerPolicy=single-numa-node``` and ```TopologyMangerPolicy=container``` configuration. ```TopologyMangerScope=pod``` is basically the same case as ```TopologyMangerPolicy=container``` but with a single container.
+
+#### Table legend:
+- R = Requested resources by the container
+- NUMA = Resources available under a NUMA.
+- R [<,=,>] NUMA = The requested resources by the container is (less/equal/more) than the resources available under the NUMA.
+- R [<,=,>] Node = The requested resources by the container is (less/equal/more) than the resources available under the Node.
+- SUM(R) = accumulate the resources of all containers 
+- Node = Resources available under the whole node.
+- N = Any number of containers
+- NVM = What ever value in the field, it won't affect the test result (a placeholder for any possible value).
+
+| test type | init containers | requests                         | containers | requests                         | state   |
+|:----------|-----------------|----------------------------------|------------|----------------------------------|---------|
+| 1         | 0               | N/A                              | 1          | R < NUMA                         | Running |
+| 2         | 0               | N/A                              | 1          | NUMA < R < Node                  | Pending |
+| 3         | 0               | N/A                              | N > 1      | R < NUMA                         | Running |
+| 4         | 0               | N/A                              | N > 1      | NUMA < SUM(R) < Node && R < NUMA | Running |
+| 5         | 0               | N/A                              | N > 1      | SUM(R) > Node && R < NUMA        | Pending |
+| 6         | 0               | N/A                              | N > 1      | NUMA < R < Node                  | Pending |
+| 7         | 1               | NUMA < R < Node                  | NVM        | NVM                              | Pending |
+| 8         | 1               | R < NUMA                         | 1          | R < NUMA                         | Running |
+| 9         | 1               | R < NUMA                         | 1          | NUMA < R < Node                  | Pending |
+| 10        | 1               | R < NUMA                         | N > 1      | R < NUMA                         | Running |
+| 11        | 1               | R < NUMA                         | N > 1      | NUMA < SUM(R) < Node && R < NUMA | Running |
+| 12        | 1               | R < NUMA                         | N > 1      | SUM(R) > Node && R < NUMA        | Pending |
```

Comment:
Sure. I removed these tests from the unit tests (but kept them in the ```TESTS.md``` since it's still a valid and possible case).
I also mentioned your note in the commit that this should be covered by the ```NodeResourceFit```.

Reply:
It would be nice to either split the table or add a column or add another table to mark which case applies to which test kind (unit, integration, e2e)